BHoM_Adapter/AdapterActions/_PushMethods/AdapterId/AssignId.cs

```diff
@@ -39,9 +41,9 @@ public abstract partial class BHoMAdapter
             bool refresh = true;
             foreach (T item in objects)
             {
-                if (AdapterIdName != null)
+                if (AdapterIdFragmentType != null)
                 {
-                    item.CustomData[AdapterIdName] = NextFreeId(typeof(T), refresh);
+                    item.SetAdapterId(AdapterIdFragmentType, NextFreeId(typeof(T), refresh));
                     refresh = false;
```

Comment:
Extra null change here:

```suggestion
                    object nextId = NextFreeId(typeof(T), refresh);
                    if (nextId != null)
                    {
                        item.SetAdapterId(AdapterIdFragmentType, nextId);
                        refresh = false;
                    }
```